src/sidebar/helpers/test/mention-suggestions-test.js

```diff
@@ -14,58 +14,107 @@ describe('combineUsersForMentions', () => {
     },
   );
 
-  it('merges, dedups and sorts users who already annotated with group members', () => {
-    const usersWhoAnnotated = [
-      {
-        userid: 'acct:janedoe@example.com',
-        username: 'janedoe',
-        displayName: 'Jane Doe',
-      },
-      {
-        userid: 'acct:cecilia92@example.com',
-        username: 'cecilia92',
-        displayName: 'Cecelia Davenport',
-      },
-    ];
-    const focusedGroupMembers = {
-      status: 'loaded',
-      members: [
+  [
+    // With username mode, list sorted by username
+    {
+      mentionMode: 'username',
+      expectedUsers: [
+        {
+          userid: 'acct:albert@example.com',
+          username: 'albert',
+          displayName: 'Albert Banana',
+        },
+        {
+          userid: 'acct:cecilia1@example.com',
+          username: 'cecilia1',
+          displayName: 'Cecelia Davenport',
+        },
+        {
+          userid: 'acct:cecilia92@example.com',
+          username: 'cecilia92',
+          displayName: 'Cecelia Davenport',
+        },
+        {
+          userid: 'acct:janedoe@example.com',
+          username: 'janedoe',
+          displayName: 'A Jane Doe',
+        },
+      ],
+    },
+    // With display-name mode, list sorted by display name, falling back to
+    // username in case of same display name
```

Comment:
We should have a test here to clarify how casing and accents are expected to be handled.